src/main/java/org/apereo/openlrs/OpenLRSAuthenticationFilter.java

```diff
@@ -111,11 +111,24 @@ private void authenticateOAuth(String authorizationHeader, HttpServletRequest re
 						normalizedParams.put(key, value);
 					}
 				}
-				
+
+				StringBuffer requestURL = request.getRequestURL();
+				String requestScheme = request.getScheme();
+				String originalScheme = request.getHeader("x-forwarded-proto");
+
+				// If original request scheme (maybe through load-balancer) is different, use it instead.
```

Comment:
To complete that idea...

The `UriComponentsBuilder.fromHttpRequest()` static factory method will make a builder from a request object and apply the forwarding info headers, but it requires a `HttpRequest` object, not `HttpServletRequest` as this code is using.  I can't just cast one to the other, either.  Maybe I can create a new `HttpRequest` object from a `HttpServletRequest`, or maybe I'll try using another factory method, like `UriComponentsBuilder.fromHttpUrl()`.  Unfortunately, none of the other factory methods apply the forwarding info headers, so that's a bummer.

Also, this is from a newer version of Spring than the project is using.  It's a somewhat old version, but even the "rc1" branch is using it.  I've tried specifying a newer version of Spring, but it breaks other dependencies.  I was able to specify a newer version of `spring-web` instead of the entire framework.  That didn't break all the other dependencies and it let me discover the inability to cast the request object, but until I get past that, I'm not sure whether it breaks other things.

We'll see.